### ISSUE TITLE
Weight-at-age 1988

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [pacific-hake/hake-assessment#703]
Figure out why the age 14 is 1.45, where interpolation would be the average of (0.83+1.20)/2.

**Comment by kellijohnson-NOAA** [comment]
The value was wrongly bolded, there is is one age-14 fish from Canadian JV in 1988 that weighed 1.45 kg.

**Comment by aaronmberger-nwfsc** [comment]
I can’t remember if bold means no data or less than three?  Would one sample then still interpolate?

**Comment by kellijohnson-NOAA** [comment]
Bold is just no data, and interpolation is only done for NA values.

**Comment by kellijohnson-NOAA** [comment]
Fixed, as much as I can for the 2020 assessment and documented in SRG appendix. Changing milestone to 2021 to ensure this doesn't happen again.

**Comment by andrew-edwards** [comment]
Sounds good. So the conclusion is that there was some extra data, which changed some values, and added age-14 fish for 1988 but we didn't unbold it.

**Comment by kellijohnson-NOAA** [comment]
I am uncertain which samples to use out of the two files that I have available. @cgrandin can you help?

R-object names == File names
* __canall__ == LengthWeightAge_data_1975to2017_IncludeCanada.csv used for CA
* __candataindat__ == LWAdata_1975to2007 used for US but not CA

# Bad
* LengthWeightAge_data_1975to2017_IncludeCanada.csv (i.e., canall) has duplicated Canadian data for CAN_shoreside in 2004 and 2005. The data appears to be the exact same, it is just that the duplicated values have more significant digits.
* LengthWeightAge_data_1975to2017_IncludeCanada.csv is missing 1988 Joint-Venture samples (n = 334)
* LengthWeightAge_data_1975to2017_IncludeCanada.csv is missing 1990 Joint-Venture samples are different b/c 136 samples in June
* LWAdata_1975to2007 has 172 JV samples in 1990 and that are not in canall
* Number of acoustic samples for Can is different between the files for every year that there are samples, except for 2001 where there are 396 samples in both files.

# Good
* LengthWeightAge_data_1975to2017_IncludeCanada.csv is missing 107 weights from CAN_domestic or CAN_shoreside from March of 2007. I think these might have been intentionally excluded b/c I remember Chris telling me that they were pretty early in the fishery.

**Comment by cgrandin** [comment]
I did find the issue from last year regarding this: https://github.com/pacific-hake/hake-assessment/issues/494

The file I edited to contain "Final" Canadian weights in 2019 is here: https://drive.google.com/drive/folders/1LMysb83wJKae_6axDKMm0N1l6E1A9MgM

**BUT...** I think it was a bit of a hack job and I can't remember what I did exactly so I decided to redo all of them from scratch and made this new file: https://drive.google.com/drive/folders/1zJb49WuUBxYq42jM4mGgeSMMIPMfPyuX

This contains all weight-age data we have in the database including 1985-2020. There aren't any duplicated data this time and all samples are included as CAN_commercial. I don't think there is any need to separate them into JV, shoreside, and FT. Also I set Sex to "U" for all of them because I didn't include the sex in the query output I have. If we need the sex I'll have to go into the office and re-write and re-run the query.

I don't remember the 2007 issue and I don't see any reference to it in the issues!

**Comment by kellijohnson-NOAA** [comment]
Questions:
1. How come there are no weights from the Canadian survey in
  * 2005
  * 2007
  * 2019
2. Do we want to include samples from Feb and March, where they were previously excluded? Seems like a good idea to keep them in if they contribute to the catches and the age data. I cannot remember why we excluded them before.

**Comment by cgrandin** [comment]
Well 2019 is because I didn't extract them, so I'll have to go into the office for that. I booked a spot so I can do that tomorrow. So, tomorrow I'll see what happened with the 2005 and 2007 data as well and double check all the survey data. The reason for excluding Feb and March was that it was likely Strait of Georgia catch, but I will do a more thorough examination tomorrow on that as well. There will be a better way of filtering those out I'm sure. My suspicion is that none of that will make any difference anyway but to be thorough I should make sure it's all in there and set up procedures/code to do this properly in the future.

**Comment by cgrandin** [comment]
I extracted all the weight/age data again for both shoreside and acoustic survey. I changed the filter on the shoreside to actual areas instead of the first three months. I also found that there were erroneous records included in early years, which were extracted before I came along. So some years are now missing and others are included, E.g. 1985 is gone and 1982 and 1990 are included. I double-checked these and indeed there are no records with both weight and age for 1985 and the other years which were removed.

I put the file on the Google drive: https://drive.google.com/drive/folders/1zJb49WuUBxYq42jM4mGgeSMMIPMfPyuX?tid=0B5lSVg5Eq86uQllqNmdmczFVU2c. I changed it to an RDS file because it's less than 10% the size of the CSV.
